lib/serviceDiscovery/engines/consul/index.js

```diff
@@ -242,7 +241,7 @@ class ConsulService extends Service {
 				logger.error.data(err).log(`Failed to renew session ${this._sessionId}`);
 			}
 
-			this._keepaliveTimer = setTimeout(this._keepalive.bind(this, sessionId), this.ttl);
+			this._keepaliveTimer = setTimeout(this._keepalive.bind(this, sessionId), this.ttl / 2);
```

Comment:
Shouldn't we change the default ttl value instead? See: https://github.com/mage/mage/blob/master/lib/serviceDiscovery/engines/consul/index.js#L16

Reply:
The default TTL doesn't matter so much, we want to make sure that we're actually hitting Consul well before the TTL would expire.  Doing it at the TTL interval means if any network hiccup hits we're at risk of the service 'going down' and this whole issue rearing up again.

Specifically, from Consul docs:

> When a TTL is being used, clients should be aware of clock skew issues: namely, time may not progress at the same rate on the client as on the Consul servers. It is best to set conservative TTL values and to renew in advance of the TTL to account for network delay and time skew.

Reply:
OK - let's try it with half-time then.